packages/broker/src/plugins/operator/MaintainTopologyService.ts

```diff
@@ -16,72 +16,51 @@ function toStreamIDSafe(input: string): StreamID | undefined {
     }
 }
 
-function singletonSet<T>(element: T): Set<T> {
-    return new Set<T>([element])
-}
+// function singletonSet<T>(element: T): Set<T> {
+//     return new Set<T>([element])
+// }
 
 export class MaintainTopologyService {
     private readonly streamrClient: StreamrClient
-    private readonly operatorClient: OperatorClient
+    private readonly maintainTopologyHelper: MaintainTopologyHelper
     private readonly subscriptions = new Multimap<StreamID, Subscription>()
     private readonly synchronizer = new SetMembershipSynchronizer<StreamID>()
     private readonly concurrencyLimit = pLimit(1)
 
-    constructor(streamrClient: StreamrClient, operatorClient: OperatorClient) {
+    constructor(streamrClient: StreamrClient, operatorClient: MaintainTopologyHelper) {
         this.streamrClient = streamrClient
-        this.operatorClient = operatorClient
+        this.maintainTopologyHelper = operatorClient
     }
 
     async start(): Promise<void> {
-        this.operatorClient.on('addStakedStream', this.onAddStakedStream)
-        this.operatorClient.on('removeStakedStream', this.onRemoveStakedStream)
-        await this.operatorClient.start()
-        const rawStreamIds = await this.operatorClient.getStakedStreams()
-        const blockNumber = 10 // TODO: fix properly
-        const streamIds = new Set(compact([...rawStreamIds].map(toStreamIDSafe)))
-        const { added } = this.synchronizer.ingestSnapshot(streamIds, blockNumber)
-        for (const streamId of added) {
-            await this.addStream(streamId, blockNumber)
-        }
-        logger.info('Started')
+        logger.info('Starting MaintainTopologyService')
+        this.maintainTopologyHelper.on('addStakedStream', this.onAddStakedStreams)
+        this.maintainTopologyHelper.on('removeStakedStream', this.onRemoveStakedStream)
+        await this.maintainTopologyHelper.start()
+        logger.info('Started MaintainTopologyService')
     }
 
-    // eslint-disable-next-line class-methods-use-this
     async stop(): Promise<void> {
-        this.operatorClient.stop()
+        this.maintainTopologyHelper.stop()
         logger.info('stopped')
     }
 
-    private onAddStakedStream = this.parseStreamIdWrapper(async (streamId: StreamID, blockNumber: number) => {
-        const { added } = this.synchronizer.ingestPatch(singletonSet(streamId), 'added', blockNumber)
-        if (added.length !== 1) {
-            logger.warn('Ignore already subscribed stream', { streamId, blockNumber })
-            return
-        }
-        if (this.subscriptions.get(streamId).length > 0) {
-            logger.warn('Ignore already subscribed stream', { streamId, blockNumber })
-            return
-        }
-        await this.addStream(streamId, blockNumber)
-    })
+    private onAddStakedStreams = async (streamIDs: string[]) => {
+        streamIDs.map(this.parseStreamIdWrapper(this.addStream.bind(this)))
```

Comment:
Method `parseStreamIdWrapper` probably needs rework to simplify this, but I can look at that in a follow-up PR.